python/ray/data/dataset.py

```diff
@@ -1354,17 +1355,52 @@ def to_spark(self,
         core_worker = ray.worker.global_worker.core_worker
         locations = [
             core_worker.get_owner_address(block)
-            for block in self.get_blocks()
+            for block in self.get_internal_block_refs()
         ]
         return raydp.spark.ray_dataset_to_spark_dataframe(
-            spark, self.schema(), self.get_blocks(), locations)
+            spark, self.schema(), self.get_internal_block_refs(), locations)
 
-    def to_pandas(self) -> List[ObjectRef["pandas.DataFrame"]]:
+    def to_pandas(self, limit: int = 1000) -> "pandas.DataFrame":
```

Comment:
this might be controversial: if user only specify `to_pandas()` without specifying limit, and the result is truncated, should we warn user as this doesn't seem intuitive behavior?

Reply:
Seems like Clark already has the logger.warn() below.